src/message/pack_encrypted/mod.rs

```diff
@@ -129,6 +134,72 @@ impl Message {
 
         Ok((msg, metadata))
     }
+
+    fn validate(&self, to: &str, from: Option<&str>, sign_by: Option<&str>) -> Result<bool> {
+        if !is_did(to) {
+            Err(err_msg(
+                ErrorKind::IllegalArgument,
+                format!("`to` value is not a valid DID of DID URL: {}", to),
+            ))?;
+        }
+
+        match from {
+            Some(from) if !is_did(from) => Err(err_msg(
+                ErrorKind::IllegalArgument,
+                format!("`from` value is not a valid DID of DID URL: {}", from),
+            ))?,
+            _ => {}
+        }
+
+        match sign_by {
+            Some(sign_by) if !is_did(sign_by) => Err(err_msg(
+                ErrorKind::IllegalArgument,
+                format!(
+                    "`sign_from` value is not a valid DID of DID URL: {}",
+                    sign_by
+                ),
+            ))?,
+            _ => {}
+        }
+
+        let (to_did, _) = did_or_url(to);
+
+        let self_to = &self.to;
+        match self_to {
+            Some(self_to) if !self_to.contains(&to_did.into()) => {
+                Err(err_msg(
+                    ErrorKind::IllegalArgument,
+                    format!(
+                        "`message.to` value {:?} does not contain `to` value's DID {}",
+                        self_to, to_did
+                    ),
+                ))?;
+            }
+            _ => {}
+        }
+
+        match from {
+            Some(from) => {
+                let (from_did, _) = did_or_url(from);
+                let self_from = &self.from;
+                match self_from {
+                    Some(self_from) if from_did != self_from => {
+                        Err(err_msg(
+                            ErrorKind::IllegalArgument,
+                            format!(
```

Comment:
I suggest don't put parameters to formatted error messages. They can be just not good looking or readable strings that can confuse users or developer. 

There is better way to get value of parameter in code that calls us.